terraform/aws/modules/drbd_node/main.tf

```diff
@@ -46,6 +46,7 @@ resource "aws_instance" "drbd" {
   vpc_security_group_ids      = [var.security_group_id]
   availability_zone           = element(var.availability_zones, count.index)
   source_dest_check           = false
+  user_data                   = templatefile("${path.root}/cloudadmin.tpl", { publickey = var.common_variables["public_key"] })
```

Comment:
Why not to have the username itself as a variable in the template? (maybe as we do not need it...)

```
  user_data                   = templatefile("${path.root}/cloudadmin.tpl", 
{
 username = var.authorized_user,
 publickey = var.common_variables["public_key"]
 })
  ```
  
  And the template like
  ```
  #cloud-config to deploy cloudadmin user in EC2 instances
cloud_final_modules:
- [users-groups,always]
users:
  - name: ${username}
    groups: [ wheel ]
    sudo:
      - "ALL=(ALL) NOPASSWD:ALL"
    shell: /bin/bash
    ssh-authorized-keys: 
    - ${publickey}
  ```

Reply:
Good idea. Didn't think of that as the goal was to add `cloudadmin`, but if it's a variable it can be overwritten via tfvars file or terraform command line. Will do that in a new PR.